ledger-api/grpc-definitions/com/daml/ledger/api/v1/admin/object_meta.proto

```diff
@@ -0,0 +1,51 @@
+// Copyright (c) 2022 Digital Asset (Switzerland) GmbH and/or its affiliates. All rights reserved.
+// SPDX-License-Identifier: Apache-2.0
+
+syntax = "proto3";
+
+package com.daml.ledger.api.v1.admin;
+
+option java_outer_classname = "ObjectMetaOuterClass";
+option java_package = "com.daml.ledger.api.v1.admin";
+option csharp_namespace = "Com.Daml.Ledger.Api.V1.Admin";
+
+// Represents metadata corresponding to some participant resource (e.g. a participant user).
+//
+// Based on 'ObjectMeta' meta used in Kubernetes API.
+// See https://github.com/kubernetes/apimachinery/blob/master/pkg/apis/meta/v1/generated.proto#L640
+message ObjectMeta {
+
+  // An opaque value populated by the participant.
+  //
+  // You may use it for concurrent change detection by means of optimistic locking.
+  // To do that include it unmodified in a change request.
+  // 'PRECONDITION_VIOLATED' error is signaled when 'resourceVersion' mismatch is detected.
+  //
+  // If you leave it empty no concurrent change detection checks are applied.
+  //
+  // Optional
+  string resourceVersion = 6;
```

Comment:
TODO: Clarify that the optimistic locking pertains to the resource the ObjectMeta is embedded in (e.g. a User with a `ObjectMeta metadata` field)